lib/puppet/parser/functions/extname.rb

```diff
@@ -0,0 +1,18 @@
+#
+# extname.rb
+#
+module Puppet::Parser::Functions
```

Comment:
Please implement using the 4.x function API and namespace the function to stdlib